pkg/progress/progress.go

```diff
@@ -83,18 +89,26 @@ func (m *Manager) UpdateProgressRemaining(progress string, remaining float64) {
 		if p.total < remaining {
 			p.total = remaining
 		}
-		if p.lastTimeRemaining < remaining {
-			p.lastTimeRemaining = remaining
+
+		if p.history.Len() > p.historyLen {
```

Comment:
```suggestion
		if p.history.Len() >= p.historyLen {
```

Reply:
This is on purpose because we will put one record when initialization.

Reply:
So the actually length limit of p.history is `historyLen+1`?

Reply:
> So the actually length limit of p.history is `historyLen+1`?

IMO, `historyLen` is used for time window. For example, if updateInterval is equal to one minute, then historyLen is equal to 10, so it takes 11 points

Reply:
change it to windowLengthLimit